### PR TITLE
Build/installation instructions

### PR DESCRIPTION
**Pull request by aprokop**
Looking for a very coarse feedback:
- What do we want here?
- What do we not want here?
- Should we talk about containers?

### CODE REVIEW

File: docs/build.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+## Manual installation
+
+ArborX has the following dependencies:
+
+Dependency | Version | Required
+--- | --- | ---
+Kokkos | 2.8.00 | :heavy_check_mark:
+MPI    | 2      |
+CMake  | 3.14   | :heavy_check_mark:
```
Comment (masterleinad) [inline docs/build.md]: I would also add `Boost` and `benchmark` even if it's just for testing and examples.

Comment (aprokop) [reply docs/build.md]: Examples do not need `boost`. Benchmarks are not for users, I think. So I would argue against complicating this.

File: docs/build.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+The dependencies may be installed using [Spack](https://github.com/spack/spack) package manager:
+FIXME (try out Kokkos install)
+```shell
+spack install kokkos
+spack install mpi
+```
+or manually (cmake and mpi - easy, Kokkos is here):
+FIXME: do we need instructions to install Kokkos here?
```
Comment (masterleinad) [inline docs/build.md]: I would link to `https://github.com/kokkos/kokkos/blob/develop/BUILD.md` or similar.

File: docs/build.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    -D BUILD_SHARED_LIBS=ON
+
+    ### TPLs ###
+    -D ARBORX_ENABLE_MPI=ON
+    -D CMAKE_PREFIX_PATH="$KOKKOS_DIR"
+
+    ### COMPILERS AND FLAGS ###
+    -D CMAKE_CXX_COMPILER="nvcc_wrapper"
+    )
+cmake "${ARGS[@]}" "${ARBORX_DIR}"
```
Comment (masterleinad) [inline docs/build.md]: Not sure about that. Once `Kokkos 3.0` is released, finding `Kokkos` should be easier. I would likely only list the `CMake` options and mention the compiler and `Kokkos`. Apart from that everything is the usual `CMake` stuff.

File: docs/build.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+```
+If building with CUDA enabled ArborX/Kokkos, add
+```
+-D CMAKE_CXX_COMPILER="nvcc_wrapper"
```
Comment (masterleinad) [inline docs/build.md]: That's already in there.

File: docs/build.md (hunk 1 above)
Comment (aprokop) [reply docs/build.md]: Question to all: if we don't use boost for core and examples, what is the minimum version of CMake that we need?

File: docs/build.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    -D CMAKE_BUILD_TYPE=RELWITHDEBINFO
+    -D CMAKE_INSTALL_PREFIX="$ARBORX_INSTALL_DIR"
+    -D BUILD_SHARED_LIBS=ON
+
+    ### TPLs ###
+    -D ARBORX_ENABLE_MPI=ON
+    -D CMAKE_PREFIX_PATH="$KOKKOS_DIR"
+
+    ### COMPILERS AND FLAGS ###
+    -D CMAKE_CXX_COMPILER="nvcc_wrapper"
```
Comment (aprokop) [inline docs/build.md]: Should this be removed in favor of the comment below?

File: docs/build.md (hunk 1 above)
Comment (dalg24) [reply docs/build.md]: I don't know but we only require 3.12 in any case

Comment (dalg24) [reply docs/build.md]: Also we may use a different marker to signify required for unit tests.

Comment (aprokop) [reply docs/build.md]: @dalg24 1. Where is 3.12 requirement coming from? 2. I don't think we need to talk about boost at all.

**Comment by aprokop** [comment]
Updated.

File: docs/build.md (hunk 1 above)
Comment (dalg24) [reply docs/build.md]: https://github.com/arborx/ArborX/blob/d41aabc39a0cacc00c9ef9a44f85f145b3c796ba/CMakeLists.txt#L1

**Comment by aprokop** [comment]
Are we missing anything important here?

**Comment by aprokop** [comment]
@sslattery There was one originally (see the `Update` patch), but it seems that it just invites a lot more questions. I don't have a strong opinion here, though.

**Comment by sslattery** [comment]
I agree that it invites more questions - is the CMake here standard enough that a Kokkos build and an ArborX build straightforward enough to not warrant additional instruction?

**Comment by aprokop** [comment]
Kokkos is definitely tricky. There is no CMake proper build right now, and I just use `generate_makefile.sh` with all the custom options. ArborX is pretty standard and that is why I specified just the differences in few bullets.

**Comment by masterleinad** [comment]
I don't see too much of a reason to explicitly talk about `Kokkos`' `generate_makefile.sh` when we are going to move to the next release soonish anyway.

**Pull request merged**